src/client/content_cao.cpp

```diff
@@ -1131,6 +1135,19 @@ void GenericCAO::step(float dtime, ClientEnvironment *env)
 		rot_translator.val_current = m_rotation;
 		updateNodePos();
 	}
+
+	if (m_animated_meshnode) {
+		// Everything must be updated; the whole transform
+		// chain as well as the animated mesh node.
+		// Otherwise, bone attachments would be relative to
+		// a position that's one frame old.
+		if(m_matrixnode){
```

Comment:
Irrlicht does this before render, but by that time it's already too late since we're animating the model right here. Note that we only evaluate transforms if we have an animated mesh node, other kinds of CAOs will not take this code path. We'll have to eat the occasional redundant evaluate until a more robust transform system is implemented.